dev/PushNotifications/PushNotificationsLongRunningTask/ForegroundSinkManager.cpp

```diff
@@ -17,19 +19,46 @@ void ForegroundSinkManager::Remove(std::wstring const& appId)
     m_foregroundMap.erase(appId);
 }
 
-bool ForegroundSinkManager::InvokeForegroundHandlers(std::wstring const& appId, winrt::com_array<uint8_t> const& payload, ULONG const& payloadSize)
+bool ForegroundSinkManager::InvokeForegroundHandlers(
+    std::wstring const& appId,
+    winrt::com_array<uint8_t> const& payload,
+    HSTRING correlationVector,
+    ULONG const& payloadSize)
 {
     auto lock = m_lock.lock_exclusive();
 
     auto it = m_foregroundMap.find(appId);
     if (it != m_foregroundMap.end())
     {
+        // IWpnForegroundSink2 declares InvokeAll expecting a correlation vector.
+        // The SDK may not know about this interface, so we make sure
+        // if the foreground sink implements it, then route the payload appropriately.
         BOOL foregroundHandled = true;
-        if (FAILED(it->second->InvokeAll(payloadSize, payload.data(), &foregroundHandled)))
+        auto foregroundSink = it->second;
+        ComPtr<IWpnForegroundSink2> foregroundSink2;
```

Comment:
It probably is low impact, but can we use winrt::com_ptr here?

Reply:
No, because the foreground sinks are stored as ComPtrs in the list.

Reply:
Unless we change the list to use winrt, but I think it's out of scope.